### DISCUSSION TITLE
undefined reference to `Os::Mutex::lock()' - ComponentBase::loadParameters()

### DISCUSSION THREAD

**Discussion by arturodrt** [nasa/fprime#1917] [Q&A] [answered]
Hello, I am working on writing unit tests for an fprime component. I have worked with unit tests for active components without any problem. This time, my component is passive and I added a parameter of type U32 to the fpp model.

When I run the **fprime-util build --ut** command it fails and I get the following error.

I am not sure what I am missing here. Any ideas of what may be causing this error?
![mutex error](https://user-images.githubusercontent.com/63877617/222276540-e6295037-30be-405b-96d3-f8de54db2cb4.jpg)

**Comment by LeStarch** [comment, marked as answer]
You likely need to add:

```
set(UT_MOD_DEPS
    Os
)
```
Before your call to `register_fprime_ut()` in the `CMakeLists.txt` file.

**Reply by arturodrt** [reply]
That solved the issue! Thank you very much!
